File: task2.rb
# Task 2
require 'date'

month = {"января"    => 1, 
         "февраля"   => 2, 
         "марта"     => 3, 
         "апреля"    => 4, 
         "мая"       => 5, 
         "июня"      => 6, 
         "июля"      => 7, 
         "августа"   => 8, 
         "сентября"  => 9, 
         "октября"   => 10, 
         "ноября"    => 11, 
         "декабря"   => 12}  

current_date = DateTime.new(ARGV[2].to_i, month[ARGV[1]], ARGV[0].to_i)
new_year_date = DateTime.new(current_date.year, 12, 31)
result = new_year_date - current_date

puts result.to_i

File: task2.js
'use strict'
let month = {"января": 0,
             "февраля": 1,
             "марта": 2,
             "апреля": 3,
             "мая": 4,
             "июня": 5,
             "июля": 6,
             "августа": 7,
             "сентября": 8,
             "октября": 9,
             "ноября": 10,
             "декабря": 11};

let currentDate = new Date(parseInt(process.argv[4]), month[process.argv[3]], parseInt(process.argv[2]));
let newYearDate = new Date(currentDate.getFullYear(), 11, 31);
let result = (newYearDate - currentDate) / 1000 / 60 / 60 / 24;

process.stdout.write(String(result));
